### PR TITLE
!per +act #17842 move BackoffSupervisor to akka.pattern

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/akka/akka/issues/17842

### CODE REVIEW

File: akka-docs/rst/general/supervision.rst [hunk 1]
```diff
@@ -199,6 +199,25 @@ external resource, which may also be one of its own children. If a third party
...
+This pattern is useful when the started actor fails because some external resource is not available,
+and we need to give it some time to start-up again. One of the prime examples when this is useful is
+when a :ref:`PersistentActor <persistence-scala>` fails with an persistence failure - which indicates that
+the database may be down or overloaded, in such situations it makes most sense to give it a little bit of time
+to recover before the peristent actor is restarted.
+
+The following snippet shows how to create a backoff supervisor which will start the given echo actor
+in increasing intervals of 3, 6, 12, 24 and finally 30 seconds:
+
+.. includecode:: ../scala/code/docs/pattern/BackoffSupervisorDocSpec.scala#backoff
```
Comment (ktoso) [inline akka-docs/rst/general/supervision.rst]: So this doc is a "general" doc, so lang agnostic, however showing how to use this seemed useful. Do we need to split up into java/supervision and scala/supervision or create a new page for that pattern? 

cc @2m @drewhk @viktorklang

Comment (drewhk) [inline akka-docs/rst/general/supervision.rst]: I think you can link two code examples, i.e. one for Java and Scala.

File: akka-actor/src/main/scala/akka/pattern/BackoffSupervisor.scala [hunk 2]
```diff
@@ -24,8 +25,8 @@ object BackoffSupervisor {
    *   started again, if it is terminated
    * @param maxBackoff the exponential back-off is capped to this duration
    * @param randomFactor after calculation of the exponential back-off an additional
-   *   random delay based on this factor is added, e.g. 0.2 adds up to 20%
-   *   delay
+   *   random delay based on this factor is added, e.g. `0.2` adds up to `20%` delay.
```
Comment (drewhk) [inline akka-actor/src/main/scala/akka/pattern/BackoffSupervisor.scala]: it is highly encouraged to add randomization. This should be mentioned in the docs.

**Comment by drewhk** [comment]
LGTM otherwise

File: akka-docs/rst/general/supervision.rst (hunk 1 above)
Comment (ktoso) [inline akka-docs/rst/general/supervision.rst]: Yeah, sounds good

File: akka-actor/src/main/scala/akka/pattern/BackoffSupervisor.scala (hunk 2 above)
Comment (ktoso) [inline akka-actor/src/main/scala/akka/pattern/BackoffSupervisor.scala]: done, added encouragement in docs

**Pull request merged**